classes/local/bigbluebutton/recordings/recording.php

```diff
@@ -103,10 +111,11 @@ public static function read($id) {
      * CRUD read by indicated attributes.
      *
      * @param array $attributes
+     * @param bool $fetchremote are we fetching remote recording info too ?
      *
      * @return [stdClass] one or many bigbluebuttonbn_recordings records indexed by recordingid.
      */
-    public static function read_by($attributes) {
+    public static function read_by($attributes, $fetchremote = true) {
```

Comment:
The recording should be an intermediate object, therefore it should have a HAS_A relationship with both the moodle mdl_bigbluebuttonbn_recordings.instance and the bigbluebutton recording object (converted either to an object or an array). 

Therefore there is always a fetch, regardless if the recording is taken from the cache or it is actually fetched from BBB

Reply:
Yes, but in my case I needed method to get info from the database.
BTW: why do we need the full recording in $object->recording ? Could we just get the values we need instead of the full object ?